compiler/src/iree/compiler/Dialect/Stream/Analysis/ResourceUsage.cpp

```diff
@@ -484,6 +484,30 @@ class ValueResourceUsage : public AbstractResourceUsage<DFX::ValueElement> {
             getState() ^= resultUsage.getState();
           }
         })
+        .Case<IREE::Stream::AsyncExecuteOp, IREE::Stream::AsyncConcurrentOp>(
+            [&](auto op) {
+              IREE::Stream::AsyncConcurrentOp c;
+              // Take on the state from the internal usage.
+              for (auto yieldOp :
+                   op.getClosureBodyRegion()
+                       .template getOps<IREE::Stream::YieldOp>()) {
+                auto &yieldUsage = solver.getElementFor<ValueResourceUsage>(
+                    *this,
+                    Position::forValue(
+                        yieldOp.getOperand(result.getResultNumber())),
+                    DFX::Resolution::REQUIRED);
+                getState() ^= yieldUsage.getState();
```

Comment:
The operator overload caught me for a minute there

Reply:
yeah, funky stuff I have to look up each time (it's all based on the LLVM Attributor style) :P